### PR TITLE
Translated all code to Kotlin

### PR DESCRIPTION
**Pull request by vshtuka**

### CODE REVIEW

File: app/src/main/java/com/example/superimagesviewer/MosaicFragment.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    private fun initRecyclerView() {
+        val layoutManager = GridLayoutManager(context, GRID_SPAN_COUNT,
+                GridLayoutManager.VERTICAL, false)
+        layoutManager.spanSizeLookup = object : SpanSizeLookup() {
+            override fun getSpanSize(position: Int): Int {
+                return calculateGridCount(position)
+            }
+        }
+        recyclerView!!.setHasFixedSize(true)
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/MosaicFragment.kt]: Do you know difference between someVar!!. and someVar?.

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import android.app.Application
+import androidx.lifecycle.MutableLiveData
+import android.graphics.drawable.Drawable
+import androidx.lifecycle.LiveData
+import android.os.AsyncTask
+import java.io.IOException
+import java.lang.ref.WeakReference
+
+class MosaicRepository(application: Application) {
+    val mosaicsList = MutableLiveData<List<Drawable>>()
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: Please make it private

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    private val drawables = mutableListOf<Drawable>()
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: Please format according to a codestyle

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    fun getMosaicsList(): LiveData<List<Drawable>> {
+        return mosaicsList
+    }
+
+    private inner class InsertAsyncTask(application: Application) : AsyncTask<Void, Void, Void>() {
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: Please avoid usage of AsyncTask, use kotlin coroutine instead

File: app/src/main/java/com/example/superimagesviewer/MosaicFragment.kt (hunk 1 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/MosaicFragment.kt]: someVar!!. asserts that the expression is not null, someVar?. is safe call operator that checks that the value to the left of it is not null

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt (hunk 2 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: done

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt (hunk 3 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: done

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt (hunk 4 above)
Comment (vshtuka) [reply app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: done

File: app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        withContext(Dispatchers.Main) {
+            mosaicsList.postValue(drawables)
+        }
+    }
+
+    companion object {
+        private const val IMAGES_FOLDER_NAME = "MyImages"
+    }
+
+    init {
```
Comment (FAQEnD) [inline app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt]: init should be always on top, like a constructor

**Review by FAQEnD — APPROVED** [review]

**Pull request merged**